test/ParticleGeneratorTest.cpp

```diff
@@ -0,0 +1,33 @@
+//
+// Created by cpsch on 22.05.2025.
+//
+
+#include "ParticleGeneratorTest.h"
+#include <gtest/gtest.h>
+
+
+
+TEST(SimpleCube, ParticleGeneratorTests)
+{
+ ParticleContainer con= ParticleGenerator::generateParticleContainer({0.0,0.0,0.0},
+  {2,2,2},1.0,1.0,{0.0,0.0,0.0},brown);
+ EXPECT_TRUE(ParticleContainer::cont_sem_eq(con,cube1));
+ for (auto i=cube1.begin(); i != cube1.end(); ++i)
+ {
+  std::cout << (*i).toString() << std::endl;
```

Comment:
```suggestion
  std::cout << i->toString() << std::endl;
```
just a style suggestion, and I would also recommend in general using a more descriptive name for the iterator, it helps improve code readability